lib/OptimizationOptimJL/src/OptimizationOptimJL.jl

```diff
@@ -14,6 +14,7 @@ SciMLBase.requiresbounds(opt::Optim.SAMIN) = true
 SciMLBase.supports_opt_cache_interface(opt::Optim.AbstractOptimizer) = true
 SciMLBase.supports_opt_cache_interface(opt::Union{Optim.Fminbox, Optim.SAMIN}) = true
 SciMLBase.supports_opt_cache_interface(opt::Optim.ConstrainedOptimizer) = true
+SciMLBase.requiresgradient(Optim.AbstractOptimizer) = !(opt isa Optim.ZerothOrderOptimizer)
```

Comment:
Looks good, but this wrapper will need more of the traits set to true (Newton, NewtonTrustRegion and IPNewton)

Reply:
And not all of them follow the abstract type, especially check the SAMIN and FminBox algorithms